tests/src/test/scala/org/http4s/headers/KeepAliveSuite.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s
+package headers
+
+import org.http4s.laws.discipline.ArbitraryInstances._
+
+class KeepAliveSuite extends HeaderLaws {
+
+  checkAll("Keep-Alive", headerLaws[`Keep-Alive`])
+
+  test("invalid (empty) Keep-Alives should result in failure") {
+    assertEquals(
+      `Keep-Alive`(None, None, List.empty),
+      ParseResult.fail("Invalid Keep-Alive header", "All fields of Keep-Alive were empty")
+    )
+  }
+
+  test("parse keep-alive with only timeout") {
+    assertEquals(
+      Header[`Keep-Alive`].parse("timeout=3"),
+      `Keep-Alive`(Some(3), None, List.empty)
+    )
+  }
+
+  test("parse keep-alive with only max") {
+    assertEquals(
+      Header[`Keep-Alive`].parse("max=33"),
+      `Keep-Alive`(None, Some(33), List.empty)
+    )
+  }
+
+  test("parse keep-alive with timeout and max") {
+    assertEquals(
+      Header[`Keep-Alive`].parse("timeout=3, max=33"),
+      `Keep-Alive`(Some(3), Some(33), List.empty)
+    )
+  }
+
+  test("parse keep-alive with timeout, max, and extensions with quoted string") {
+    assertEquals(
+      Header[`Keep-Alive`].parse("""timeout=3, max=33, extKey="il""""),
+      `Keep-Alive`(Some(3), Some(33), List(("extKey", Some("il"))))
+    )
+  }
+
+  test(
+    "parse keep-alive with timeout, max, and extensions with 3 keys (extKey as a token) some have values and one does not") {
+    assertEquals(
+      Header[`Keep-Alive`].parse("""timeout=3, max=33, extKey=il, nextKey="pi", thirdKey="""),
+      `Keep-Alive`(
+        Some(3),
+        Some(33),
+        List(("extKey", Some("il")), ("nextKey", Some("pi")), ("thirdKey", None)))
```

Comment:
I think this should be `Some("")`.  Don't you want to test `thirdKey` without the equals sign?

Reply:
You found a bug!

Reply:
This should be fixed here now:  https://github.com/http4s/http4s/pull/5165/commits/a36b6cd9fb80306acb8b776270d59df6562094c4

and in https://github.com/http4s/http4s/pull/5165/commits/a0cc9c13ef0cc298919a63d588611267ffdbd7a9 to fix the tests.  Doing "nextKey=" is now invalid as it should have been before and would result in a parse failure.